runtime/pkg/rduckdb/catalog.go

```diff
@@ -0,0 +1,253 @@
+/*
+Example init logic:
+- Sync remote files with the local cache
+- Create a catalog
+- Traverse the local files and call addTableVersion for table
+Example write logic:
+- Call addTableVersion after adding a new table version
+- Call removeTable when deleting a table
+Example read logic:
+- Call acquireSnapshot when starting a read
+- If it doesn't already exist, create a schema for the snapshot ID with views for all the table version in the snapshot
+- Call releaseSnapshot when done reading the snapshot
+Example removeFunc logic:
+- Detach the version
+- Remove the version file
+- If there are no files left in it, remove the table folder
+*/
```

Comment:
Probably want to remove this comment or provide the info with more context elsewhere